### PR TITLE
Crush:add some safe judgment.

### PR DESCRIPTION
**Pull request by songbaisen**
Crush: add some safe judgment.

Fixes: #14496
Signed-off-by: song baisen song.baisen@zte.com.cn

### CODE REVIEW

File: src/crush/builder.c [hunk 1]
```diff
@@ -681,7 +681,9 @@ crush_make_bucket(struct crush_map *map,
 		return (struct crush_bucket *)crush_make_straw_bucket(map, hash, type, size, items, weights);
 	case CRUSH_BUCKET_STRAW2:
 		return (struct crush_bucket *)crush_make_straw2_bucket(map, hash, type, size, items, weights);
-	}
+	default:
+               return NULL;
+       }
 	return 0;
```
Comment (liewegas) [inline src/crush/builder.c]: This doesn't really change anything; just switch return 0 to return NULL.

File: src/crush/mapper.c [hunk 2]
```diff
@@ -73,6 +73,7 @@ int crush_find_rule(const struct crush_map *map, int ruleset, int type, int size
 static int bucket_perm_choose(struct crush_bucket *bucket,
 			      int x, int r)
 {
+       assert(bucket->size != 0);
```
Comment (liewegas) [inline src/crush/mapper.c]: this isn't useful since we'd crash on the next line anyway and get the same stack trace.  but it does slow us down slightly.

**Comment by liewegas** [comment]
The first assert is good, the others aren't helpful.  Can you always put a space after the colon in your commit messages?

**Comment by songbaisen** [comment]
@liewegas  Thanks for your review.I have changed it.

**Pull request merged**